src/modules-by-example.adoc

```diff
@@ -0,0 +1,320 @@
+local symbol table only via `imports` and no `symbols` field.
+
+Let's look more closely at the definition of `local`:
+
+[{nrm}]
+----
+(*module* local
+  (*symbols* ["s1", "s2"]))
+----
+
+The `*module*` keyword starts an S-expression that defines a new module with the given name.
+The `*symbols*` keyword starts a subform that defines the module's exported symbol table.
+This clause expects a list of strings, and both the syntax and semantics are the same as the
+`symbols` field of `$ion_shared_symbol_table`.
+
+Once this module is defined, we can _install_ it into the local symbol table:
+
+[{nrm}]
+----
+  install_symbols: [ local ]
+----
+
+This field expects a list of symbols that match names declared in the `modules` field.  The
+resulting local symbol table is simply the concatenation of the exported symbol tables of those
+modules.  This works the same way as the `imports` field of `$ion_symbol_table`.
+
+
+=== Importing Symbols
+
+Given the equivalencies above, we could perform a naive round-trip of the preceding 1.1 document
+back to 1.0. First, turn the `local` module into the equivalent shared symbol table:
+
+[{nrm}]
+----
+**$ion_shared_symbol_table**::{
+  name: "local",
+  version: 1,
+  symbols: ["s1", "s2"]
+}
+----
+
+Then translate `install_symbols:[local]` into its 1.0 equivalent:
+
+[{nrm}]
+----
+*$ion_1_0*
+**$ion_symbol_table**::{
+  imports: [{ name: "local", version: 1, max_id: 2 }]
+}
+----
+
+NOTE:  Even ignoring Ion 1.1, this is how you would hoist local symbols into a
+new shared symbol table.
+
+The latter imports-only document has this 1.1 equivalent:
+
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [
+    (*import* local "local" 1 2)
+  ],
+  install_symbols: [ local ]
+}
+----
+
+Here we see a new form inside the `modules` field that imports a module into the encoding
+environment and assigns it a name.
+The `*import*` keyword starts an S-expression that expects three or four arguments. The first is
+a symbolic name that we can use later to refer to the imported module.  The remaining arguments
+are effectively the `name`, `version` and `max_id` fields of the 1.0 `imports` struct.
+
+TIP: From the perspective of Ion 1.1, shared symbol tables _are_ encoding modules.
+
+
+=== Declaring Multiple Modules
+
+Within a directive's `modules` field, both `*module*` and `*import*` clauses have the same
+ultimate effect, declaring a name->module mapping for the encoding environment.
+Let’s look at a scenario with both imported and locally-defined symbols:
+
+[{nrm}]
+----
+*$ion_1_0*
+**$ion_symbol_table**::{
+  imports: [{ name: "shared1", version: 1, max_id: 10 },
+            { name: "shared2", version: 2, max_id: 20 }],
+  symbols: ["s1", "s2"]
+}
+----
+
+Here’s the Ion 1.1 equivalent in terms of symbol allocation order:
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [(*import* n1 "shared1" 1 10),
+            (*import* n2 "shared2" 2 20),
```

Comment:
Is `n1`/`n2` required?  Can they be omitted if `shared1`/`shared2` are valid module names?

Reply:
The symbols are required, and are local names.  The second argument must be a string, just as in an `import` struct.

Reply:
Can it not be required--to avoid DRY when possible? `(import "shared1" 1 10)` => `(import shared1 "shared1" 1 10)`, I mean not a big deal, but a little annoying.

Reply:
In most cases, symbol table names are long and include qualifying prefixes, since its a global "namespace".  I think few will be usable as identifying symbols.

The syntax could infer the symbolic name, but it only works if the string satisfies the content constraint (ie it only uses identifier characters).  That doesn't work universally, and it's complex to explain, and the resulting schema is irregular in how the local module-names are expressed... so I'm not inclined.

Reply:
I've submitted #209 which uses more realistic SST names.